broker-daemon/relayer/market-watcher.js

```diff
@@ -0,0 +1,126 @@
+const EventEmitter = require('events')
+const { MarketEvent } = require('../models')
+const { migrateStore } = require('../utils')
+
+/**
+ * @class Watch a relayer market and put the events into a data store
+ * It emits two events:
+ * - `sync` when it is caught up to the market
+ * - `end` when it is no longer connected to the market
+ */
+class MarketWatcher extends EventEmitter {
+  /**
+   * Set up the watcher
+   * @param  {grpc.ServerStreaming} watcher        Stream of events from the Relayer
+   * @param  {Sublevel}             store          Leveldb compatible store
+   * @param  {Object}               RESPONSE_TYPES Response types from the proto file
+   * @param  {Object} logger
+   * @return {MarketWatcher}
+   */
+  constructor (watcher, store, RESPONSE_TYPES, logger) {
```

Comment:
should we just call this `stream` instead of watcher?

Reply:
@treygriffith Wondering if there is a more descriptive term for this object `watcher`